src/main/kotlin/org/jitsi/nlj/stats/PacketDelayStats.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.nlj.stats
+
+import org.jitsi.nlj.PacketInfo
+import java.util.concurrent.atomic.AtomicLong
+
+class DelayStats {
+    private val totalDelayMs = AtomicLong(0)
+    private val totalPackets = AtomicLong(0)
+    val averageDelay: Double
+        get() = totalDelayMs.get() / totalPackets.get().toDouble()
+    val maxDelayMs = AtomicLong(0)
+
+    fun addPacket(packetInfo: PacketInfo) {
+        val delayMs = if (packetInfo.receivedTime > 0) {
+            System.currentTimeMillis() - packetInfo.receivedTime
+        } else {
+            -1
+        }
+        if (delayMs >= 0) {
+            totalDelayMs.addAndGet(delayMs)
+            if (delayMs > maxDelayMs.get()) {
+                maxDelayMs.set(delayMs)
```

Comment:
You could use `maxDelayMs.updateAndGet(prev -> max(prev, delayMs))` to make this atomic

Reply:
I wonder if there's an easy way of avoiding creating a lambda on every call though

Reply:
left this as is for now (i don't know of a way to do that and avoiding the lambda)

Reply:
For later: you can use `org.jitsi.util.MonotonicAtomicLong` (it doesn't use `updateAndGet` in order to work on java7, but a side effect is that it doesn't create an extra lambda). If we decide to do that we should move it to the jitsi-utils package.